howto/anbox/develop-addon.md

```diff
@@ -0,0 +1,46 @@
+Developing and testing addons using the Anbox Management Service (AMS) may be time-consuming. Instead, a container with `--devmode` enabled can provide a safe environment to test addons and their scripts without having to upload the addon to the AMS.
+
+This guide explains how to use a container in development mode to develop and test an addon using Anbox runtime. See [development mode](https://discourse.ubuntu.com/t/17763#dev-mode) to learn more about development mode enabled containers.
+
+## Launch a container in development mode
+
+Start a raw container with `--devmode` enabled:
+
+```
+amc launch --devmode --raw
+```
+
+The command prints out the ID of the container. Note down the container ID for next steps.
+
+Use the `amc shell <container_id>` command to open a shell inside the container. `container_id` is the container ID from the previous step.
+
+Alternatively, you can use `amc exec <container_id> <command_options>` to directly execute commands inside the container.
+
+## Create an addon within the container
+
+Use the container as a remote environment to develop your addon. To make your addon source available within the container, either copy the addon manifest and hooks using the [`lxc file push`](https://documentation.ubuntu.com/lxd/en/latest/howto/instances_access_files/#push-files-from-the-local-machine-to-the-instance) command or clone a git repository using SSH.
+
+You can test your addon scripts by running it inside the container shell. For example, `ADDON_DIR=$PWD ./hooks/install` can help test if the install hook of the addon works.
+
+To troubleshoot issues within the container, try either of the following options:
+* Run `amc logs <container-id>` on the host to see the Anbox runtime logs.
+* Run `journalctl --no-pager` within the container to view container logs.
+* Restart the container using `amc stop <container_id>` and then `amc start <container_id>`.
+
+## Example: Launch an SSH-enabled container for remote development
+
+```bash
+# Launch and obtain the container id
+id="$(amc launch --devmode -s ssh --raw)"
+# Install the ssh-import-id package
+amc exec $id -- apt install -y ssh-import-id
+# Import SSH keys for GitHub; Use lp:<user_name> for Launchpad
+amc exec $id -- ssh-import-id gh:<user_name>
+# Get the <public_address> and <node_port> values of the container
+amc show $id --format=json | json_pp
```

Comment:
what's the `json_pp`?

Reply:
It just pretty prints the json format. I chose that because the command utility is available by default on Ubuntu machines(https://perldoc.perl.org/json_pp)

Reply:
@adglkh Simon had suggested `jq` first but when I tried it out for the first time, it required `jq` to be installed. Since it is not something installed by default and assumes that on the part of the user, I used `json_pp`. If you feel strongly about this, I can change it to using `jq` with additional comment saying that it has to be installed. Please let me know.

Reply:
Okay, I see. That makes sense to me.
Thanks!

Reply:
what advantage does json_pp give us here? It looks like we only use it for pretty printing but `amc show --format=json` should pretty print already. The idea with jq was to extract the ip/node port.

Reply:
This is what I see when I try it. If you still feel `jq` should be used as it is more useful, I can update the documentation accordingly and mention that `jq` should be installed prior.

```bash
ubuntu@vm5:~$ amc show $id --format=json
{"id":"cik1pnmk9eqm4inenf7g","name":"ams-cik1pnmk9eqm4inenf7g","status":"started","node":"lxd0","created_at":"2023-07-07 14:10:38 +0000 UTC","application":{},"image":{"id":"cgrqjd6k9eqlsruefcng","version":1},"network":{"address":"x.x.x.x","public_address":"x.x.x.x","services":[{"port":22,"node_port":10001,"expose":false,"protocols":["tcp"],"name":"ssh"}]},"stored_logs":[],"error_message":"","config":{"devmode":true},"resources":{},"app_id":"","app_version":0,"image_id":"cgrqjd6k9eqlsruefcng","image_version":1}ubuntu@vm5:~$ 
ubuntu@vm5:~$ amc show $id --format=json | json_pp
{
   "app_id" : "",
   "app_version" : 0,
   "application" : {},
   "config" : {
      "devmode" : true
   },
   "created_at" : "2023-07-07 14:10:38 +0000 UTC",
   "error_message" : "",
   "id" : "cik1pnmk9eqm4inenf7g",
   "image" : {
      "id" : "cgrqjd6k9eqlsruefcng",
      "version" : 1
   },
   "image_id" : "cgrqjd6k9eqlsruefcng",
   "image_version" : 1,
   "name" : "ams-cik1pnmk9eqm4inenf7g",
   "network" : {
      "address" : "x.x.x.x",
      "public_address" : "x.x.x.x",
      "services" : [
         {
            "expose" : false,
            "name" : "ssh",
            "node_port" : 10001,
            "port" : 22,
            "protocols" : [
               "tcp"
            ]
         }
      ]
   },
   "node" : "lxd0",
   "resources" : {},
   "status" : "started",
   "stored_logs" : []
}
ubuntu@vm5:~$ 
```

Reply:
As discussed in the SU, using `jq`.